python/PyQt6/core/auto_generated/qgsattributes.sip.in

```diff
@@ -72,14 +72,15 @@ typedef QVector<QVariant> QgsAttributes;
 
   QgsAttributes *qv = new QgsAttributes;
   Py_ssize_t listSize = PyList_GET_SIZE( sipPy );
-  qv->reserve( listSize );
+  qv->resize( listSize );
+  QVariant *outData = qv->data();
 
   for ( Py_ssize_t i = 0; i < listSize; ++i )
   {
     PyObject *obj = PyList_GET_ITEM( sipPy, i );
     if ( obj == Py_None )
     {
-      qv->append( QVariant( QVariant::Int ) );
+      *outData++ = QVariant( QVariant::Int );
```

Comment:
Are we gaining something relevant here ?

Reply:
Yes -- it reduces the qt implicitly shared container detachment checks from happening once per attribute appended to just once upfront. Ie both QVector::append and QVector::data check for shared use and detachment **every** time they are called, even if it's in a trivial loop like this. These detachment checks add up quickly!